sys/net/nfc/ndef/ndef_message.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2024 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     ndef
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the NDEF message format
+ *
+ * @author      Nico Behrens <nifrabe@outlook.de>
+ *
+ * @}
+ */
+
+#include "net/nfc/ndef/ndef.h"
+#include "log.h"
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+#include <assert.h>
+#include <inttypes.h>
+#include <byteorder.h>
+
+#define SHORT_RECORD_PAYLOAD_LENGTH_SIZE 1
+#define LONG_RECORD_PAYLOAD_LENGTH_SIZE  4
+
+/* Structure of the header byte */
+#define RECORD_MB_MASK                   0x80
+#define RECORD_ME_MASK                   0x40
+#define RECORD_CF_MASK                   0x20
+#define RECORD_SR_MASK                   0x10
+#define RECORD_IL_MASK                   0x08
+#define RECORD_TNF_MASK                  0x07
+
+size_t ndef_get_size(const ndef_t *ndef)
+{
+    return ndef->buffer.cursor - ndef->buffer.memory;
+}
+
+void ndef_pretty_print(const ndef_t *ndef)
+{
+    printf("----------------\n");
+    printf("\n");
+    for (size_t i = 0; i < (size_t)ndef->record_count; ++i) {
+        const ndef_record_desc_t *record = &ndef->records[i];
+        printf("Record %d\n", i);
+        printf("----\n");
+        printf("MB: %d\n", (record->header[0] & RECORD_MB_MASK) != 0);
+        printf("ME: %d\n", (record->header[0] & RECORD_ME_MASK) != 0);
+        printf("CF: %d\n", (record->header[0] & RECORD_CF_MASK) != 0);
+        printf("SR: %d\n", (record->header[0] & RECORD_SR_MASK) != 0);
+        printf("IL: %d\n", (record->header[0] & RECORD_IL_MASK) != 0);
+        printf("TNF: %d\n", record->header[0] & RECORD_TNF_MASK);
+        printf("Header byte: 0x%02x\n", record->header[0]);
+        printf("Type length: %d\n", record->type_length[0]);
+        printf("Type: ");
+        for (int j = 0; j < record->type_length[0]; ++j) {
+            printf("%c", record->type[j]);
+        }
+        printf("\n");
+
+        printf("Payload length size: %u\n", record->record_type);
+        uint32_t payload_length;
+        if (record->record_type == NDEF_SHORT_RECORD) {
+            payload_length = record->payload_length[0];
+        }
+        else {
+            payload_length = record->payload_length[0] << 24 |
+                             record->payload_length[1] << 16 |
+                             record->payload_length[2] << 8 |
+                             record->payload_length[3];
```

Comment:
```suggestion
            payload_length = ((uint32_t) record->payload_length[0]) << 24 |
                             ((uint32_t) record->payload_length[1]) << 16 |
                             ((uint32_t) record->payload_length[2]) <<  8 |
                             ((uint32_t) record->payload_length[3]);
```

This would fix this CI error:
```
ndef_message.c: In function ‘ndef_pretty_print’:
ndef_message.c:76:56: error: left shift count >= width of type [-Werror=shift-count-overflow]
             payload_length = record->payload_length[0] << 24 |
                                                        ^
ndef_message.c:77:56: error: left shift count >= width of type [-Werror=shift-count-overflow]
                              record->payload_length[1] << 16 |
```


Currently you're trying to left-shift an `uint8_t` (which is the type of `*(record->payload_length)`) and the compiler won't implicitly cast it to 32 bits *before* doing the shift.